### PR TITLE
Added more checks on nice_address_set_from_string (fixes #1973)

### LINKED ISSUE #1973: Invalid ICE address displays error messages from libnice
I have crafted an invalid ICE candidate (IP "99192.168.3.12"):

```
Mar 02 15:39:08 [Mon Mar  2 15:39:08 2020] {"candidate":{"candidate":"candidate:2895048534 1 udp 2122260223 99192.168.3.12 23128 typ host generation 0 ufrag lpF3 network-id 1","sdpMLineIndex":0,"sdpMid":"audio"},"handle_id":7528623161552875,"session_id":2146165875570853,"transaction":"QvoFVvL7T1SX","janus":"trickle"}
Mar 02 15:39:08 [Mon Mar  2 15:39:08 2020] [WSS-0x7f42b0005fa0] Done, parsing message: 279 bytes
Mar 02 15:39:08 [Mon Mar  2 15:39:08 2020] Got a Janus API request from janus.transport.websockets (0x7f42b0005e70)
Mar 02 15:39:08 [Mon Mar  2 15:39:08 2020] [7528623161552875] Trickle candidate (audio): candidate:2895048534 1 udp 2122260223 99192.168.3.12 23128 typ host generation 0 ufrag lpF3 network-id 1
Mar 02 15:39:08 [Mon Mar  2 15:39:08 2020] [7528623161552875]  Adding remote candidate component:1 stream:1 type:host 99192.168.3.12:23128
Mar 02 15:39:08 [Mon Mar  2 15:39:08 2020] [7528623161552875]  Transport: UDP
Mar 02 15:39:08 (process:1288): libnice-CRITICAL **: file address.c: line 147 (nice_address_set_port): should not be reached
Mar 02 15:39:08 [Mon Mar  2 15:39:08 2020] [7528623161552875]    Candidate added to the list! (1 elements for 1/1)
Mar 02 15:39:08 [Mon Mar  2 15:39:08 2020] [7528623161552875] ICE already started for this component, setting candidates we have up to now
Mar 02 15:39:08 [Mon Mar  2 15:39:08 2020] [7528623161552875] ## Setting remote candidates: stream 1, component 1 (1 in the list)
Mar 02 15:39:09 [Mon Mar  2 15:39:08 2020] [7528623161552875] >> Remote Stream #1, Component #1
Mar 02 15:39:09 (process:1288): libnice-CRITICAL **: file address.c: line 249 (nice_address_to_string): should not be reached
Mar 02 15:39:09 (process:1288): libnice-CRITICAL **: file address.c: line 165 (nice_address_get_port): should not be reached
```

There is several error messages rated as "critical" by libnice. 

The error comes from the result of `nice_address_set_from_string` that is not checked.
https://github.com/meetecho/janus-gateway/blob/2d3c53e8a874c5717126af7319c19c64ae3854a0/sdp.c#L787
https://github.com/meetecho/janus-gateway/blob/2d3c53e8a874c5717126af7319c19c64ae3854a0/sdp.c#L792

If `nice_address_set_from_string` fails (the IP is invalid), I suggest to have a specific log message (level LOG_ERR maybe).

### PR DESCRIPTION
**Pull request by mp16**
More checks

### CODE REVIEW

File: sdp.c [hunk 1]
```diff
@@ -796,11 +796,15 @@ int janus_sdp_parse_candidate(void *ice_stream, const char *candidate, int trick
 				c->password = g_strdup(stream->rpass);
 				if(c->type == NICE_CANDIDATE_TYPE_SERVER_REFLEXIVE || c->type == NICE_CANDIDATE_TYPE_PEER_REFLEXIVE) {
 					added = nice_address_set_from_string(&c->base_addr, rrelip);
-					nice_address_set_port(&c->base_addr, rrelport);
+					if(added) {
+						nice_address_set_port(&c->base_addr, rrelport);
+					}
```
Comment (lminiero) [inline sdp.c]: No need for the brackets here, since the action is a single line, please remove them.

File: sdp.c [hunk 2]
```diff
@@ -796,11 +796,15 @@ int janus_sdp_parse_candidate(void *ice_stream, const char *candidate, int trick
...
 				} else if(c->type == NICE_CANDIDATE_TYPE_RELAYED) {
 					/* FIXME Do we really need the base address for TURN? */
 					added = nice_address_set_from_string(&c->base_addr, rrelip);
-					nice_address_set_port(&c->base_addr, rrelport);
+					if(added) {
+						nice_address_set_port(&c->base_addr, rrelport);
+					}
```
Comment (lminiero) [inline sdp.c]: See above.

**Comment by mp16** [comment]
I think it's ok for the code style and the CLA.

**Comment by lminiero** [comment]
Thanks, merging then :+1:

**Pull request merged**